src/main/java/meditracker/command/AddCommand.java

```diff
@@ -52,17 +74,88 @@ public AddCommand(String arguments) throws MediTrackerException {
      *
      * @param medicationManager      The MedicationManager object representing the list of medications.
      * @param dailyMedicationManager The DailyMedicationManager object representing the list of daily medications.
+     * @throws NullPointerException   if any of the required objects are null.
+     * @throws NumberFormatException  if there is an error in parsing numeric values.
      */
     @Override
     public void execute(MedicationManager medicationManager,
-                        DailyMedicationManager dailyMedicationManager) {
-        Medication medication = new Medication(medicationName, medicineQuantity, medicineDosage, expiryDate, intakeFreq,
-                remarks);
+                        DailyMedicationManager dailyMedicationManager) throws NullPointerException,
+            NumberFormatException {
+        setMedicineAttributes();
+        Medication medication = new Medication(medicationName, medicationQuantity, medicationDosage,
+                medicationDosageMorning, medicationDosageAfternoon, medicationDosageEvening, expiryDate,
+                intakeFreq, repeat, remarks);
+
         DailyMedication dailyMedication = new DailyMedication(medicationName);
         medicationManager.addMedication(medication);
         dailyMedicationManager.addDailyMedication(dailyMedication);
+        assertionTest(medicationManager, dailyMedicationManager);
+        Ui.showAddCommandMessage();
+    }
+
+    /**
+     * Sets the medication attributes based on parsed arguments.
+     *
+     * @throws NumberFormatException if there is an error in parsing numeric values.
+     * @throws NullPointerException  if any of the required arguments are null.
+     */
+    private void setMedicineAttributes() throws NumberFormatException, NullPointerException {
+        medicationName = parsedArguments.get(ArgumentName.NAME);
+        expiryDate = parsedArguments.get(ArgumentName.EXPIRATION_DATE);
+        intakeFreq = parsedArguments.get(ArgumentName.INTAKE_FREQUENCY);
+        remarks = parsedArguments.get(ArgumentName.REMARKS);
+        repeat = parsedArguments.get(ArgumentName.REPEAT);
+
+        String medicationQuantity = parsedArguments.get(ArgumentName.QUANTITY);
+        String medicationDosage = parsedArguments.get(ArgumentName.DOSAGE);
+        String medicationDosageMorning = parsedArguments.get(ArgumentName.DOSAGE_MORNING);
+        String medicationDosageAfternoon = parsedArguments.get(ArgumentName.DOSAGE_AFTERNOON);
+        String medicationDosageEvening = parsedArguments.get(ArgumentName.DOSAGE_EVENING);
+
+        parseStringToValues(medicationQuantity, medicationDosage, medicationDosageMorning, medicationDosageAfternoon,
+                medicationDosageEvening);
+    }
```

Comment:
Might be better to just create and return a `Medication` object here. This will help to reduce the need to store them as attributes of the class as the lifetime of these variables just have to be scoped to this method.

Reply:
Thanks Wen En. Apologies for my poor understanding but could you elaborate further? The rationale of storing the attributes first is so that they could be passed together into the Medication constructor together to create the Medication object. 

Or are you suggesting that I should create my Medication object here and return it at the end?

Reply:
Yep, I meant to say create the Medication object in this method and return it at the end

Reply:
Resolved in https://github.com/AY2324S2-CS2113T-T09-1/tp/pull/75/commits/694dc8f7241818e259fe1e7c99d10c14ad7c07b5